### PR TITLE
Introduce Request Feedback

### LINKED ISSUE #43: Implement Loading/Error Handling
### Problem
When a user clicks `Save` it triggers a request but there's no feedback to the user. 

### Proposed Solution
Let's add a loading state that goes away when we receive a `200` from the network request. While we're at it, let's add graceful Error Handling for anything not in the 200 range.

The solution must be something that can be implemented on several screens so code or widget must be agnostic and not tied to one screen.

### PR DESCRIPTION
**Pull request by cbhernan**
### What does this PR do?
Introduces a loading screen for when requests are in flight and displays a Snackbar with a message relaying the success or failure of the request.

### Which issue(s) is/are related to this PR?
Closes #43

### CODE REVIEW

File: lib/controllers/api_implementation.dart [hunk 1]
```diff
@@ -83,18 +83,25 @@ class UserApi extends Api {
 
     try {
       final response = await http.post(
-          Uri.parse('/$url'),
+          Uri.parse('/updateUser'),
           headers: headers,
           body: body
-      );
+      ).timeout(const Duration(seconds: 15));
```
Comment (Paz-X) [inline lib/controllers/api_implementation.dart:89]: I was testing updating the user without internet to see if the error snackbar would pop up. But instead it just kept loading. Let's have the loading disappear if it ends up timing out and verify the timeout is in fact working.

**Review by Paz-X — CHANGES_REQUESTED** [review]
Let's verify the timeout

File: lib/controllers/api_implementation.dart (hunk 1 above)
Comment (cbhernan) [reply lib/controllers/api_implementation.dart:89]: That's what this line does.

**Review by nickchaula — APPROVED** [review]
approved

**Pull request merged**